metrics.go

```diff
@@ -0,0 +1,106 @@
+	return &Metrics{
+		// Interception-related metrics.
+
+		// Estimated cardinality: 2 providers, 5 models, 2 statuses, 2 routes, 3 methods = up to 120 PER INITIATOR.
+		InterceptionCount: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Subsystem: "interceptions",
+			Name:      "total",
+			Help:      "The count of intercepted requests.",
+		}, append(baseLabels, "status", "route", "method", "initiator_id")),
+		// Estimated cardinality: 2 providers, 5 models, 2 routes = up to 20.
+		// NOTE: route is not unbounded because this is only for intercepted routes.
+		InterceptionsInflight: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+			Subsystem: "interceptions",
+			Name:      "inflight",
+			Help:      "The number of intercepted requests which are being processed.",
+		}, append(baseLabels, "route")),
+		// Estimated cardinality: 2 providers, 5 models, 7 buckets + 3 extra series (count, sum, +Inf) = up to 100.
+		InterceptionDuration: promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
+			Subsystem: "interceptions",
+			Name:      "duration",
+			Help: "The total duration of intercepted requests. " +
+				"The majority of this time will be the upstream processing of the request. " +
+				"aibridge has no control over upstream processing time, so it's just an illustrative metric.",
+			// TODO: add docs around determining aibridge's *own* latency with distributed traces
+			//       once https://github.com/coder/aibridge/issues/26 lands.
+			Buckets: []float64{0.5, 2, 5, 15, 30, 60, 120},
+		}, baseLabels),
+
+		// Estimated cardinality: 2 providers, 10 routes, 3 methods = up to 60.
+		// NOTE: route is not unbounded because PassthroughRoutes (see provider.go) is a static list.
+		PassthroughCount: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Subsystem: "passthrough",
+			Name:      "total",
+			Help:      "The count of requests which were not intercepted but passed through to the upstream.",
+		}, []string{"provider", "route", "method"}),
+
+		// Prompt-related metrics.
+
+		// Estimated cardinality: 2 providers, 5 models = up to 10 PER INITIATOR.
+		PromptCount: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Subsystem: "prompts",
+			Name:      "total",
+			Help:      "The number of prompts issued by users (initiators).",
+		}, append(baseLabels, "initiator_id")),
+
+		// Token-related metrics.
+
+		// Estimated cardinality: 2 providers, 5 models, 10 types = up to 100 PER INITIATOR.
+		TokenUseCount: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Subsystem: "token_usage",
+			Name:      "total",
+			Help:      "The number of tokens used by intercepted requests.",
+		}, append(baseLabels, "type", "initiator_id")),
+
+		// Tool-related metrics.
+
+		// Estimated cardinality: 2 providers, 5 models, 3 servers, 30 tools = up to 900 PER INITIATOR.
+		InjectedToolUseCount: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Subsystem: "injected_tool_usage",
```

Comment:
re usage: `injected_tool_calls_total`?

PS. It might be worth cross-checking these names with [Prometheus naming rules](https://prometheus.io/docs/practices/naming/).

Reply:
OK cool. I included "usage" to align with our data model, but it's probably more clear as `injected_tool_calls_total`.

What specifically do you think doesn't align with the naming rules?